tests/Tests.Shared/test-api.h

```diff
@@ -0,0 +1,572 @@
+                    /* TODO: don't leak. Use better storage. */ \
+                    auto factory = metaFactory.second->CreateTestFactory(*param); \
+                    auto test = factory->CreateTest(); \
+                    \
+                    WEX::Logging::Log::StartGroup(currentTest.c_str()); \
+                    _ExchangeAlternateTestName(converter.to_bytes(currentTest), true); \
+                    try \
+                    { \
+                        ::testing::TestRunner::SetUp(test); \
+                    } \
+                    catch (const ::WEX::TestExecution::VerifyFailureException&) \
+                    { \
+                    } \
+                    \
+                    try \
+                    { \
+                        ::testing::TestRunner::TestBody(test); \
+                    } \
+                    catch (const ::WEX::TestExecution::VerifyFailureException&) \
+                    { \
+                    } \
+                    \
+                    try \
+                    { \
+                        ::testing::TestRunner::TearDown(test); \
+                    } \
+                    catch (const ::WEX::TestExecution::VerifyFailureException&) \
+                    { \
+                    } \
+                    _ExchangeAlternateTestName(std::string(), true); \
+                    \
+                    WEX::Logging::Log::EndGroup(currentTest.c_str()); \
+                    \
+                    delete factory; \
+                    delete test; \
+                } \
+            } \
+            WEX::Logging::Log::Comment(L"\nEnd of parameterized subtests for " L#className L"_" L#prefix L".\n"); \
+        } \
+    };
+
+// clang-format on
+
+#ifndef GET_CURRENT_TEST_DIRECTORY
+#define GET_CURRENT_TEST_DIRECTORY                                                                    \
+    []() {                                                                                            \
+        WEX::Common::String testDeploymentPath;                                                       \
+        WEX::TestExecution::RuntimeParameters::TryGetValue(L"TestDeploymentDir", testDeploymentPath); \
+        std::wstring toReturn(testDeploymentPath);                                                    \
+                                                                                                      \
+        std::wstring_convert<std::codecvt_utf8_utf16<wchar_t>> converter;                             \
+        return converter.to_bytes(toReturn);                                                          \
+    }
+#endif
+
+#ifndef GET_TEST_FULL_NAME
+#define GET_TEST_FULL_NAME                                                                \
+    []() {                                                                                \
+        std::string alternateTestName = _ExchangeAlternateTestName(std::string(), false); \
```

Comment:
I agree. I was trying to avoid having to use a .cpp for this but I remembered that I already did that work for the logger so I *could* make this a bit nicer my having a class with more sane storage and helpers.